test-apps/app-engine-test-app/src/main/java/com/google/gcloud/tests/appengine/MainServlet.java

```diff
@@ -0,0 +1,121 @@
+
+import com.google.gcloud.AuthCredentials;
+import com.google.gcloud.bigquery.BaseTableInfo;
+import com.google.gcloud.bigquery.BigQuery;
+import com.google.gcloud.bigquery.BigQueryOptions;
+import com.google.gcloud.bigquery.DatasetInfo;
+import com.google.gcloud.bigquery.Field;
+import com.google.gcloud.bigquery.Schema;
+import com.google.gcloud.bigquery.TableId;
+import com.google.gcloud.bigquery.TableInfo;
+import com.google.gcloud.bigquery.testing.RemoteBigQueryHelper;
+import com.google.gcloud.datastore.Datastore;
+import com.google.gcloud.datastore.DatastoreOptions;
+import com.google.gcloud.datastore.DateTime;
+import com.google.gcloud.datastore.Entity;
+import com.google.gcloud.datastore.Key;
+import com.google.gcloud.datastore.KeyFactory;
+import com.google.gcloud.storage.Blob;
+import com.google.gcloud.storage.BlobId;
+import com.google.gcloud.storage.BlobInfo;
+import com.google.gcloud.storage.BucketInfo;
+import com.google.gcloud.storage.Storage;
+import com.google.gcloud.storage.StorageOptions;
+import com.google.gcloud.storage.testing.RemoteGcsHelper;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintWriter;
+import java.nio.ByteBuffer;
+import java.nio.channels.WritableByteChannel;
+
+import javax.servlet.ServletContext;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class MainServlet extends HttpServlet {
+
+  private PrintWriter responseWriter;
+
+  @Override
+  public void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+    responseWriter = resp.getWriter();
+    runBigQueryActions(BigQueryOptions.defaultInstance().service());
+    ServletContext context = getServletContext();
+    InputStream in =
+        context.getResourceAsStream("/WEB-INF/lib/my-service-account-credentials.json");
+    runDatastoreActions(
+        DatastoreOptions.builder()
+            .authCredentials(AuthCredentials.createForJson(in))
+            .projectId("your-project-id")
+            .build()
+            .service());
+    runStorageActions(StorageOptions.defaultInstance().service());
+  }
+
+  private void runBigQueryActions(BigQuery bigquery) {
+    responseWriter.println("Testing BigQuery.");
+    String datasetName = RemoteBigQueryHelper.generateDatasetName();
+    bigquery.create(DatasetInfo.builder(datasetName).build());
+    TableId tableId = TableId.of(datasetName, "my_table");
+    BaseTableInfo info = bigquery.getTable(tableId);
+    if (info == null) {
+      responseWriter.println("Creating table " + tableId);
+      Field integerField = Field.of("fieldName", Field.Type.integer());
+      bigquery.create(TableInfo.of(tableId, Schema.of(integerField)));
+    }
+    RemoteBigQueryHelper.forceDelete(bigquery, datasetName);
+    responseWriter.println("Finished BigQuery test.");
+  }
+
+  private void runDatastoreActions(Datastore datastore) {
```

Comment:
It would be good if we can avoid the copy/paste of the service interaction code.
Maybe put it in a common place and use maven [`add-source`](http://stackoverflow.com/questions/9752972/how-to-add-an-extra-source-directory-for-maven-to-compile-and-include-in-the-bui).